spring-cloud-square-retrofit-webclient/src/main/java/org/springframework/cloud/square/retrofit/webclient/WebClientCallAdapterFactory.java

```diff
@@ -125,11 +136,28 @@ else if (isFlux) {
 						httpHeaders.put(entry.getKey(), entry.getValue());
 					}
 				});
-		if (request.body() != null) {
-			// spec.body()
-			// FIXME: body
+		RequestBody requestBody = request.body();
+		if (requestBody != null) {
+			processRequestBody(request, spec, requestBody);
 		}
 		return spec;
 	}
 
+	private void processRequestBody(Request request, WebClient.RequestBodySpec spec, RequestBody requestBody) {
+		BufferedSink bufferedSink = Okio.buffer(Okio.sink(new ByteArrayOutputStream()));
```

Comment:
This looks like it works for now, but using byte arrays puts everything in memory which is worrying.

We should have a custom Sink impl that integrates with reactor so it streams ala https://github.com/CollaborationInEncapsulation/okhttp-reactor-netty/blob/master/src/main/java/reactor/okhttp/netty/RetrofitAdapterUtils.java

Reply:
ok. will take a look at it.